File: apps/dglab-gui-tauri/src/types/events.ts
/**
 * Tauri event types and names
 */

import { DeviceInfo, DeviceState } from "./device";

/** 设备状态变更事件 */
export interface DeviceStateChangedEvent {
  device_id: string;
  state: DeviceState;
}

/** 设备功率变更事件 */
export interface DevicePowerChangedEvent {
  device_id: string;
  power_a: number;
  power_b: number;
}

/** 设备信息更新事件 */
export interface DeviceInfoUpdatedEvent {
  device_id: string;
  info: DeviceInfo;
}

/** 设备电池电量更新事件 */
export interface DeviceBatteryUpdatedEvent {
  device_id: string;
  battery: number;
}

/** 设备错误事件 */
export interface DeviceErrorEvent {
  device_id: string;
  error: string;
}

/** 事件名称常量 */
export const EVENT_NAMES = {
  DEVICE_STATE_CHANGED: "device:state_changed",
  DEVICE_POWER_CHANGED: "device:power_changed",
  DEVICE_INFO_UPDATED: "device:info_updated",
  DEVICE_BATTERY_UPDATED: "device:battery_updated",
  DEVICE_ERROR: "device:error",
} as const;

File: apps/dglab-gui-tauri/src-tauri/src/events.rs
//! 设备事件定义

use serde::{Deserialize, Serialize};

use dglab_core::device::traits::DeviceInfo;
use dglab_core::device::DeviceState;

/// 设备状态变更事件
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DeviceStateChangedEvent {
    /// 设备 ID
    pub device_id: String,
    /// 新状态
    pub state: DeviceState,
}

/// 设备功率变更事件
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DevicePowerChangedEvent {
    /// 设备 ID
    pub device_id: String,
    /// 通道 A 功率
    pub power_a: u8,
    /// 通道 B 功率
    pub power_b: u8,
}

/// 设备信息更新事件
#[derive(Debug, Clone, Serialize, Deserialize)]
#[allow(dead_code)]
pub struct DeviceInfoUpdatedEvent {
    /// 设备 ID
    pub device_id: String,
    /// 设备信息
    pub info: DeviceInfo,
}

/// 设备电池电量更新事件
#[derive(Debug, Clone, Serialize, Deserialize)]
#[allow(dead_code)]
pub struct DeviceBatteryUpdatedEvent {
    /// 设备 ID
    pub device_id: String,
    /// 电池电量 (0-100)
    pub battery: u8,
}

/// 设备错误事件
#[derive(Debug, Clone, Serialize, Deserialize)]
#[allow(dead_code)]
pub struct DeviceErrorEvent {
    /// 设备 ID
    pub device_id: String,
    /// 错误信息
    pub error: String,
}

/// 事件名称常量
#[allow(dead_code)]
pub mod event_names {
    /// 设备状态变更
    pub const DEVICE_STATE_CHANGED: &str = "device:state_changed";
    /// 设备功率变更
    pub const DEVICE_POWER_CHANGED: &str = "device:power_changed";
    /// 设备信息更新
    pub const DEVICE_INFO_UPDATED: &str = "device:info_updated";
    /// 设备电池电量更新
    pub const DEVICE_BATTERY_UPDATED: &str = "device:battery_updated";
    /// 设备错误
    pub const DEVICE_ERROR: &str = "device:error";
}
